packages/react/src/components/stepper/Stepper.stories.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { useReducer } from 'react';
+
+import { Stepper } from './Stepper';
+import styles from './Stepper.module.scss';
+import { Step } from './Step';
+import { Button } from '../button';
+
+export default {
+  component: Stepper,
+  title: 'Components/Stepper',
+  decorators: [(storyFn) => <div style={{ maxWidth: '700px' }}>{storyFn()}</div>],
+  parameters: {
+    controls: { expanded: true },
+  },
+  args: {},
+};
+
+const commonReducer = (stepsTotal) => (state, action) => {
+  switch (action.type) {
+    case 'completeStep': {
+      const activeStep = action.payload === stepsTotal ? stepsTotal : action.payload + 1;
+      return {
+        activeStep,
+        states: state.states.map((stateName, index) => {
+          if (index === action.payload - 1 && index !== stepsTotal - 1) {
+            // current one but not last one
+            return 'filled';
+          }
+          if (index === action.payload) {
+            // next one
+            return 'available';
+          }
+          return stateName;
+        }),
+      };
+    }
+    case 'setActive': {
+      return {
+        activeStep: action.payload,
+        states: state.states.map((stateName, index) => {
+          if (index === action.payload - 1) {
+            return 'available';
+          }
+          return stateName;
+        }),
+      };
+    }
+    default:
+      throw new Error();
+  }
+};
+
+// args is required for docs tab to show source code
+// eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
+export const Default = (args) => {
+  const reducer = commonReducer(5);
+
+  const initialState = {
+    activeStep: 1,
+    states: ['available', 'disabled', 'disabled', 'disabled', 'disabled'],
+  };
+
+  const labels = ['Step 1', 'Step 2', 'Step 3', 'Step 4 - longer text', 'Step 5'];
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  return (
+    <div>
+      <Stepper
+        labels={labels}
+        language="en"
+        states={state.states}
+        selectedStep={state.activeStep}
+        stepsTotal={5}
```

Comment:
The stepper does this in the demo, but the logic of how and where stepper moves comes outside of the component, and can be customised in any way possible. So if users of the stepper would want it to behave otherwise, that is possible.